### ISSUE TITLE
6.0 Feedback: Protocol analysis with invalid Python API version fails without reporting to client

### ISSUE THREAD

**Issue by mcous** [Opentrons/opentrons#10733]
### Overview

I you upload a Python protocol with a correctly shaped but invalid protocol API version (e.g. 2.42), the protocol analysis will fail on the server, but the analysis resource will not be updated, so the endpoint will report "pending" forever.

### Steps To Reproduce

This simple protocol triggers the bug:

```python

from opentrons import protocol_api

metadata = {
    "apiLevel": "2.1000",
}


def run(protocol: protocol_api.ProtocolContext) -> None:
    protocol.comment("hello world")
```

**Comment by y3rsh** [comment]
Seems like a good candidate for an automated acceptance test?
1. POST the example protocol to produce the analysis failure.
2. Wait for analysis to complete
3. GET the analysis detail and validate status is correct and the right error is present.

**Comment by ecormany** [comment]
I'd say expected behavior here should be a failed analysis with an error like you get with `opentrons_simulate`: 
```
RuntimeError: API version 2.1000 is not supported by this robot software. Please either reduce your requested API version or update your robot.
```

**Comment by mikecardarelli** [comment]
There are designs for this error on the slide-out, but they haven't been implemented yet

**Comment by mikecardarelli** [comment]
See #10569 for this issue

**Comment by mcous** [comment]
Should not be closed, server bug

**Comment by bndo** [comment]
Closed by #10780
